mesonbuild/rewriter.py

```diff
@@ -685,6 +690,17 @@ def find_node(src):
             }
             self.add_info('target', target['id'], test_data)
 
+        # Sort files
+        for i in to_sort_nodes:
+            convert = lambda text: int(text) if text.isdigit() else text.lower()
+            alphanum_key = lambda key: [convert(c) for c in re.split('([0-9]+)', key)]
+            path_sorter = lambda key: ([(len(key.split('/')[idx:]) <= 1, alphanum_key(x)) for idx, x in enumerate(key.split('/'))])
```

Comment:
I think the first item in the tuple could be something like `len(key.split('/')) - 1 == idx`. I think that would be more readable.

Reply:
Not sure about this one. I think my variant is more readable because it is more clear that it checks the number of remaining elements.

Reply:
Maybe simpler? `key.count('/') >= idx` (assuming one slash at the beginning and no trailing slash, might need to adjust a bit.)

Or if using `pathlib.Path`, then check `Path.parts` property.

Reply:
Thanks, that's even simpler. I am against using the paths functions because meson only uses `/` and I don't know how this code will behave on windows.